### PR TITLE
Upgrade the nix-based development environment for ALF

### PR DESCRIPTION
**Pull request by breakds**
## Motivation

Point the nix development environment to the up-to-date [nix-alf-devenv](https://github.com/HorizonRobotics/alf-nix-devenv). This mainly includes:

1. Upgrade `pytorch` to 1.11.0
2. Upgrade `metadrive` to its latest version
3. Added `tensor-spline` to the environment

# Solution

N/A

### CODE REVIEW

File: flake.lock [hunk 1]
```diff
@@ -6,14 +6,15 @@
         "nixpkgs": [
           "nixpkgs"
         ],
+        "tensor-splines-flake": "tensor-splines-flake",
```
Comment (breakds) [inline flake.lock:9]: Please note that the update on this file is automatically generated.

File: flake.nix [hunk 2]
```diff
@@ -12,8 +12,8 @@
   };
 
   outputs = { self, nixpkgs, ... }@inputs: inputs.utils.lib.eachSystem [
-    "x86_64-linux" "i686-linux" "aarch64-linux" "x86_64-darwin"
```
Comment (le-horizon) [inline flake.nix:15]: Is this intentional?

File: flake.nix [hunk 3]
```diff
@@ -12,8 +12,8 @@
...
+    "x86_64-linux"
   ] (system: {
-    devShell = inputs.alf-devenv.devShell."${system}";
+    devShell = inputs.alf-devenv.devShells."${system}".default;
```
Comment (le-horizon) [inline flake.nix:17]: will this change affect some of the documentation?

**Review by le-horizon — APPROVED** [review]
Not sure about two things.  Otherwise look good.

File: flake.nix (hunk 3 above)
Comment (breakds) [reply flake.nix:17]: Nope this will not.

File: flake.nix (hunk 2 above)
Comment (breakds) [reply flake.nix:15]: Good point. This is intentional as I haven't tested it on MacOS X yet.

Comment (le-horizon) [reply flake.nix:15]: Ok then.  It's best to support one frequent use case than to worry about all possible uses.  Feel free to merge.

**Pull request closed without merging**